.github/workflows/ci.yml

```diff
@@ -1,6 +1,11 @@
 name: Build and test symengine
-on: [push, pull_request]
-
+on:
+  push:
+    branches:
+      - master
```

Comment:
This makes it harder to run CI on branches on my fork. We are not using branches on this repo anyway.

Reply:
Either way fine by me - it just means that for every PR currently the CI runs twice, once for the PR and once for the push (see e.g. https://github.com/symengine/symengine/pull/2045/checks)

Reply:
reverted this change: a986efb344a44a88bd542ff3aee697ade76e3cc0